### PR TITLE
Move Isolation to the consensuscommit package

### PR DESCRIPTION
**Pull request by brfrn169**
This PR moves the `Isolation` class to the `consensuscommit` package. Also, the property name is changed to `scalar.db.consensus_commit.isolation_level`. But we can still use the old property name `scalar.db.isolation_level` for backward compatibility.

### CODE REVIEW

File: core/src/main/java/com/scalar/db/api/Isolation.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package com.scalar.db.api;
 
+/** @deprecated As of release 3.5.0. Will be removed in release 4.0.0. */
+@Deprecated
 public enum Isolation {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/api/Isolation.java:5]: We don't remove this yet for backward compatibility.

File: core/src/main/java/com/scalar/db/config/DatabaseConfig.java [hunk 2]
```diff
@@ -62,7 +60,6 @@
   public static final String PASSWORD = PREFIX + "password";
   public static final String STORAGE = PREFIX + "storage";
   public static final String TRANSACTION_MANAGER = PREFIX + "transaction_manager";
-  public static final String ISOLATION_LEVEL = PREFIX + "isolation_level";
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/config/DatabaseConfig.java:65]: If we use `DatabaseConfig.ISOLATION_LEVEL`, we need to change it to `ConsensusCommitConfig.ISOLATION_LEVEL`. It's backward incompatibility but I think it's okay.

File: core/src/main/java/com/scalar/db/config/DatabaseConfig.java [hunk 3]
```diff
@@ -204,10 +196,6 @@ public int getContactPort() {
     return transactionManagerClass;
   }
 
-  public Isolation getIsolation() {
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/config/DatabaseConfig.java:207]: Ditto. We need to change `DatabaseConfig.getIsolation()` to `ConsensusCommitConfig.getIsolation()`.

File: core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitConfig.java [hunk 4]
```diff
@@ -31,27 +40,43 @@
...
   }
 
-  @Override
   protected void load() {
-    String transactionManager = getProperties().getProperty(DatabaseConfig.TRANSACTION_MANAGER);
-    if (transactionManager != null && !transactionManager.equals("consensus-commit")) {
-      throw new IllegalArgumentException(
-          DatabaseConfig.TRANSACTION_MANAGER + " should be 'consensus-commit'");
+    if (getProperties().containsValue("scalar.db.isolation_level")) {
+      LOGGER.warn(
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitConfig.java:64]: If the old property is used, show a warning message.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**